src/session/content/message_row/reply.rs

```diff
@@ -0,0 +1,288 @@
+use crate::components::MiniThumbnail;
+use adw::subclass::prelude::BinImpl;
+use glib::closure;
+use gtk::glib::{self, clone};
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, CompositeTemplate};
+use tdlib::enums::{Message as EnumMessage, MessageContent};
+use tdlib::functions;
+
+use crate::strings;
+use crate::tdlib::{Chat, ChatType, Message, MessageSender};
+use crate::utils::spawn;
+
+mod imp {
+    use once_cell::sync::Lazy;
+
+    use crate::session::content::message_row::base::MessageBaseImpl;
+
+    use super::*;
+    use std::cell::{Cell, RefCell};
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(resource = "/com/github/melix99/telegrand/ui/content-message-reply.ui")]
+    pub(crate) struct MessageReply {
+        pub(super) sender_color_class: RefCell<Option<String>>,
+        pub(super) bindings: RefCell<Vec<gtk::ExpressionWatch>>,
+        pub(super) chat: RefCell<Option<Chat>>,
+        pub(super) reply_id: Cell<i64>,
+        pub(super) is_outgoing: Cell<bool>,
+        pub(super) is_loading: Cell<bool>,
+        pub(super) show_thumbnail: Cell<bool>,
+        #[template_child]
+        pub(super) sender_label: TemplateChild<gtk::Label>,
+        #[template_child]
+        pub(super) separator: TemplateChild<gtk::Separator>,
+        #[template_child]
+        pub(super) message_label: TemplateChild<gtk::Label>,
+        #[template_child]
+        pub(super) message_thumbnail: TemplateChild<MiniThumbnail>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for MessageReply {
+        const NAME: &'static str = "MessageReply";
+        type Type = super::MessageReply;
+        type ParentType = adw::Bin;
+
+        fn class_init(klass: &mut Self::Class) {
+            Self::bind_template(klass);
+            klass.set_css_name("messagereply");
+        }
+
+        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+            obj.init_template();
+        }
+    }
+
+    impl ObjectImpl for MessageReply {
+        fn properties() -> &'static [glib::ParamSpec] {
+            static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+                vec![
+                    glib::ParamSpecObject::builder::<Chat>("chat")
+                        .read_only()
+                        .explicit_notify()
+                        .build(),
+                    glib::ParamSpecInt64::builder("reply-id")
+                        .read_only()
+                        .explicit_notify()
+                        .build(),
+                    glib::ParamSpecBoolean::builder("is-outgoing")
```

Comment:
It should not be there. If you need it for styling you should check for parent widgets instead. Like:

```css
messagebubble.outgoing messagereply separator {
  color: @accent_color;
}
```

Reply:
> It should not be there

Well, I don't know I use it anywhere. It might be a reclic from the times before message bubble